### PR TITLE
Notifications ui redesign

### PR DESCRIPTION
**Pull request by NathanielBennett**
## My Awesome Pull Request

Leicester City, still top ..
## What does this change?

Changes the design of live notifications - moves it to the left hand column. This means a slight change in the left-hand stickiness .
## What is the value of this and can you measure success?

It's still at present a 0% ab test. We wanna ramp it up for the next big boxing match when it will with  any luck get us a load of return visiotors
## Does this affect other platforms - Amp, Apps, etc?

Before: 
notifications-before

After: 

(With explainer, disappears after first follow)
explainer

(Following)
alerts-on

(Display a message when notifications are blocked)
blocked-permissions

### CODE REVIEW

**Comment by gtrufitt** [comment]
I think you probably need some padding around your image here to bring it in line with the text padding: 
image

Also, below this:
image

File: article/app/views/liveblog/liveBlogBody.scala.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
 @import conf.switches.Switches._
 
 @defining((model.article, Edition(request).timezone)) { case (article, timezone) =>
+<div class="js-notificications-blocked"></div>
```
Comment (gtrufitt) [inline article/app/views/liveblog/liveBlogBody.scala.html]: Minor sp mistake in 'notifications' here

File: article/app/views/liveblog/liveBlogBody.scala.html [hunk 2]
```diff
@@ -50,25 +51,28 @@ <h1 itemprop="headline" class="content__headline js-score">@Html(article.trail.h
...
-                            <div class="blog__timeline blog__dropdown js-live-blog__key-events modern-visible js-updates-desktop">
-                                <div class="blog__timeline-container js-live-blog__timeline-container" data-component="timeline">
-                                    @fragments.dropdown("Key events", Some("key-events"), true) {
-                                        <ul class="timeline js-live-blog__timeline u-unstyled">
-                                            @views.html.liveblog.keyEvents("/" + article.metadata.id, KeyEventData(article.content.fields.blocks, timezone))
-                                        </ul>
-                                    }
-                                </div>
+                        <div class="js-live-blog__sticky-components">
+                            <div class="js-live-blog__sticky-components-container blog__sticky-components-container">
```
Comment (gtrufitt) [inline article/app/views/liveblog/liveBlogBody.scala.html]: I would expect the two classes to be identical except for the `js-` prefix? Though this looks like it's a fallout from previous naming so might be a major change :/

File: article/app/views/liveblog/liveBlogBody.scala.html [hunk 3]
```diff
@@ -50,25 +51,28 @@ <h1 itemprop="headline" class="content__headline js-score">@Html(article.trail.h
...
+                                            }
+                                        </div>
+                                    </div>
+                                }
+                                <div class="js-live-notifications live-notifications" data-component="notifications"></div>
                             </div>
-                        }
-                        <div class="js-components-container"></div>
+                        </div>
+                    <div class="js-components-container"></div>
```
Comment (gtrufitt) [inline article/app/views/liveblog/liveBlogBody.scala.html]: Minor: Needs another indent

File: static/src/inline-svgs/notifications-explainer-desktop.svg [hunk 4]
```diff
@@ -1 +1 @@
...
...sible" fill-rule="evenodd" clip-rule="evenodd" fill="#767676"/><path fill="#FFF" d="M118.9 107.9c-4 2.9-6.5 7.4-6.5 12.9 0 6.5 3.4 10.9 8.2 14.1l.2.2c-2.8 1.9-8.5 6.4-8.5 11.5 0 3.9 2.5 7.6 7.9 8.8-5.6 1.1-11.9 3.6-11.9 9.9v.5c-6.3-6.5-10.4-15.2-10.4-25.1.1-14.7 8.9-27 21-32.8zm28.9 56.1c0 3.9-2.9 6.9-13.5 6.9-9.3 0-12.5-3.2-12.5-6.7 0-2.6 1.1-5.8 5.7-5.8h13.3c4.6-.1 7 2.2 7 5.6zm11.4 2.3c1.2-2.5 2-5.4 2-8.6 0-9.8-5.2-14.1-16.3-14.1h-17.8c-1.9 0-3.5-1.6-3.5-3.3 0-1.2 1-2.6 2.3-3.6 2.6.9 4.9 1 8.1 1 12.7 0 21.4-6.2 21.3-17-.1-6.1-2.7-10.5-7-13.1 12.7 5.5 21.8 18.1 21.8 33.1-.1 10-4.3 19-10.9 25.6zm-18.7-45.5c0 9.8-2.9 10.9-6.5 10.9-3.5 0-6.5-.8-6.5-10.9s2.9-11.4 6.5-11.4c3.7 0 6.5 1.6 6.5 11.4z"/><text transform="translate(212 139.953)" fill="#333" font-family="'TS3TextSans'" font-size="16">Obama plan for trade deals with Asia and</text><text transform="translate(212 157.953)" fill="#333" font-size="16">Europe in tatters after Senate vote</text><text transform="translate(212 115.953)" fill="#333" font-family="'TS3TextSans'" font-size="20">The Guardian</text><text transform="translate(212 180)" fill="#BCBCBC" font-family="'TS3TextSans'" font-size="14">theguardian.com</text></g></svg>
```
Comment (gtrufitt) [inline static/src/inline-svgs/notifications-explainer-desktop.svg]: Looks like you need to run this through SVGO? Also need to remove the `xmlns` attributes.

File: static/src/inline-svgs/notifications-explainer-mobile.svg [hunk 5]
```diff
@@ -1 +1 @@
...
...108-4.024-.994-7.517-2.618-10.447-1.623-2.946-3.886-5.328-6.761-7.056 4.257 1.884 8.18 4.282 11.744 7.223 3.564 2.967 6.643 6.347 9.197 10.19 2.569 3.82 4.58 7.983 6.02 12.507 1.435 4.512 2.171 9.277 2.171 14.294zm-38.91 23.727c2.886 0 5.143.692 6.75 2.066 1.64 1.379 2.451 3.192 2.451 5.44 0 2.657-1.214 4.84-3.655 6.6-2.445 1.76-7.24 2.64-14.388 2.64-6.257 0-10.61-.923-13.05-2.73-2.436-1.825-3.65-3.934-3.65-6.3 0-1.76.52-3.482 1.585-5.189 1.059-1.69 3.112-2.527 6.117-2.527h17.84zm-29.321-67.587c-2.741 2.007-4.88 4.46-6.38 7.352-1.51 2.892-2.252 6.208-2.252 9.974 0 4.4 1 8.162 3.01 11.28 2 3.154 4.638 5.665 7.885 7.549l.381.182c-1.01.606-2.128 1.481-3.38 2.538a48.412 48.412 0 0 0-3.768 3.584 19.958 19.958 0 0 0-3.096 4.325 10.562 10.562 0 0 0-1.237 4.98c0 2.64.86 5.075 2.564 7.345 1.666 2.264 4.338 3.756 7.971 4.529-1.886.354-3.751.864-5.633 1.497a21.492 21.492 0 0 0-5.09 2.452 12.356 12.356 0 0 0-3.66 3.761c-.94 1.497-1.403 3.316-1.403 5.447v.75c-4.257-4.372-7.638-9.432-10.143-15.136-2.515-5.704-3.762-11.912-3.762-18.55 0-4.893.688-9.572 2.075-14.025 1.376-4.427 3.305-8.558 5.826-12.31a52.299 52.299 0 0 1 8.837-10.172c3.364-3.01 7.127-5.441 11.255-7.352z" fill="#005689"/></g></svg>
```
Comment (gtrufitt) [inline static/src/inline-svgs/notifications-explainer-mobile.svg]: ditto with SVGO

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 6]
```diff
@@ -43,37 +48,65 @@ define([
             return modules.getReg().then(function (reg) {return reg.pushManager.getSubscription();});
         },
 
-        configureSubscribeTemplate: function () {
-            var subscribed = modules.checkSubscriptions(),
-                hasNoSubscriptions = modules.subscriptionsEmpty(),
+        init: function() {modules.configureSubscribeButton();
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: `modules.configureSubscribeButton();` should be a new line?

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 7]
```diff
@@ -43,37 +48,65 @@ define([
...
+            if(!modules.hasSubscribed() && !modules.hasDismissedExplainer()) {
+                modules.showExplainer();
+            }
+        },
+
+        configureSubscribeButton: function () {
+            var $follow = bonzo(qwery('.js-live-notifications')[0]),
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: We're already importing `$` here so you can just use `$('.js-live-notifications')`. You don't need the `[0] ` as `.html` is a bonzo method and you're already using `[0]` in the `.one` method already.

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 8]
```diff
@@ -43,37 +48,65 @@ define([
...
-                    text: subscribed ? 'Unfollow' : 'Follow story',
-                    imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'notification-explainer']),
-                    imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'notification-explainer'])
+                src = template(followLink, {
+                    subscribed: subscribed,
+                    icon: svgs(subscribed ? 'notificationsOff' : 'notificationsOn')
                 });
 
+            $follow.html(src);
+            bean.one($follow[0], 'click', '.js-notifications__button', handler );
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Why are we using `one` here? Can you not unsubscribe and then subscribe again? If so, you would need to checkSubscriptions in your handler, not just on page load.

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 9]
```diff
@@ -43,37 +48,65 @@ define([
...
+            var src = template(explainer,{
+                closeIcon : svgs('closeCentralIcon'),
+                imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'live-notifications-explainer-svg']),
+                imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'live-notifications-explainer-svg'])
+            });
+
             fastdom.write(function () {
-                $('.js-notifications').prepend(src);
-                bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+                var notifications = $('.js-live-notifications');
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Missing `$` prefix on var name as per general consistency of code base.

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 10]
```diff
@@ -43,37 +48,65 @@ define([
...
+                notifications.append(src);
+                bean.one(notifications[0], 'click', '.js-live-notifications__item__close', function() {
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: You probably don't need two levels of nesting in the BEM classname here? Will `.js-live-notifications__item__close` suffice?

File: static/src/javascripts/bootstraps/enhanced/notifications.js (hunk 8 above)
Comment (NathanielBennett) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Yeah but each time you un or resubscribe the handler you need to use changes.

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 11]
```diff
@@ -43,37 +48,65 @@ define([
...
+            });
         },
 
         subscribeHandler: function () {
-            modules.subscribe().then(modules.follow);
+            modules.subscribe().then(modules.follow)
+                .catch( function () {
+                    if (Notification.permission === 'denied') {
+                        modules.notificationsDeniedMessage();
+                    } });
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Minor `});` should be next line?

File: static/src/javascripts/projects/common/views/ui/notifications-explainer.html [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<div class="js-live-notifications-explainer live-notifications-explainer">
+    <div class="live-notifications-explainer__content">
+        <div class="live-notifications-explainer__header">
+                <div class="live-notifications-explainer__headline">Get alerts and stay updated</div>
+                <div class="live-notifications-explainer__header-blurb">Receive notifications on your device when important updates on this story are published</div>
+        </div>
+        <div class="live-notifications-explainer__item-options">
+            <button class="js-live-notifications__item__close button button--tertiary u-faux-block-link__promote"
+                    aria-label="Dismiss"
+                    data-link-name="close button"><%=closeIcon%>
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/views/ui/notifications-explainer.html]: 👍 I like the use of `aria-label` over hidden text here

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 13]
```diff
@@ -1,20 +1,239 @@
...
+
+    .notifications__button--subscribed {
+        color: $background;
+        background-color: $foreground;
+        .inline-icon {
+            fill: $background;
+            background-color: $foreground;
+        }
+
+        &:hover {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Don't forget to `&:focus`

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 14]
```diff
@@ -1,20 +1,239 @@
...
+
+    .notifications__button--unsubscribed {
+        color: $foreground;
+        background-color: transparent;
+        .inline-icon {
+            fill: $foreground;
+            background-color: $background;
+        }
+
+        &:hover {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: `&:focus`

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 15]
```diff
@@ -1,20 +1,239 @@
...
+
+.live-notifications {
+    margin-bottom: $gs-baseline /2;
+    position: relative;
+}
+
+.notifications__button {
+
+    .inline-icon {
+        svg {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Don't need to nest these, `.notifications__button .inline-icon svg` is clearer

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 16]
```diff
@@ -1,20 +1,239 @@
...
+            width: 1.143em;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Why `em`?

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 17]
```diff
@@ -1,20 +1,239 @@
...
+    position: fixed;
+    display: block;
+    top: .625rem;
+    left: 5.275rem;
+    width: 26.825rem;
+    height: 4.7625rem;
+    background: fade-out($live-default, 1 - .95);
+    transition: background-color .25s ease;
+    filter: alpha(opacify=50);
+    -moz-opacity: .5;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Don't need filter, -moz or -khtml here - you don't have any `opacity` declaration though?

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 16 above)
Comment (NathanielBennett) [inline static/src/stylesheets/module/preferences/_notifications.scss]: I ripped the style off the breaking news close button ..

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 18]
```diff
@@ -1,20 +1,239 @@
...
+@include notifications-button-toned(dead, $live-default, $neutral-8);
+
+.live-notifications-denied {
+    position: fixed;
+    display: block;
+    top: .625rem;
+    left: 5.275rem;
+    width: 26.825rem;
+    height: 4.7625rem;
+    background: fade-out($live-default, 1 - .95);
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Confused as to why `1 - .95` and not just `.05`? @sndrs I think this came from breaking-news, what was thinking of writing it like that?

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 19]
```diff
@@ -1,20 +1,239 @@
...
+    transition: background-color .25s ease;
+    filter: alpha(opacify=50);
+    -moz-opacity: .5;
+    -khtml-opacity: .5;
+    z-index: 15000;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Should add a zindex value here to avoid zindex chaos https://github.com/guardian/frontend/blob/notifications-ui-redesign/static/src/stylesheets/base/_zindex.scss

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 20]
```diff
@@ -1,20 +1,239 @@
...
+        padding: 0;
+        width: $live-notifications-icon-width;
+        height: $live-notifications-icon-width;
+        border-color: #ffffff;
+
+        @include mq(tablet) {
+            border-color: colour(live-mute);
+        }
+
+        &:hover {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: `focus`

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 21]
```diff
@@ -1,20 +1,239 @@
...
 
-    .button--follow {
-        margin-bottom: $gs-baseline;
+        &:hover {
+            border-color: #ffffff;
+        }
     }
 }
 
+.live-notifications-denied::after {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Consistency wise, we generally use single colon for `:after`

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 22]
```diff
@@ -1,20 +1,239 @@
...
+    content: '';
+    display: inline-block;
+    width: 0;
+    height: 0;
+    border-top: 0;
+    position: absolute;
+    left: 3%;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: As your popup has a fixed position, I would fix this left in pixels so that the width of the pop-up can change but the arrow position won't

File: static/src/inline-svgs/notifications-explainer-desktop.svg (hunk 4 above)
Comment (sndrs) [inline static/src/inline-svgs/notifications-explainer-desktop.svg]: svgo will be applied automatically though

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 23]
```diff
@@ -1,20 +1,239 @@
...
+
+    .inline-marque-36 {
+        position: absolute;
+        left: 0 - $live-notifications-icon-width - $gs-gutter / 2;
+        top: 0;
+    }
+}
+
+.live-notifications-denied__header {
+    @include fs-textSans(3);;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Double `::`

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 18 above)
Comment (sndrs) [inline static/src/stylesheets/module/preferences/_notifications.scss]: i have no idea, that looks weird to me!

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 24]
```diff
@@ -1,20 +1,239 @@
...
+
+.live-notifications-explainer__header-blurb {
+    @include fs-textSans(3);
+    margin-top: $gs-baseline /2;
+}
+
+.live-notifications-explainer__headline {
+    @include fs_headline(2);
+    color: #ffffff;
+    font-weight: 500;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Shouldn't have to pull in a specific font-weight here I don't believe?

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 25]
```diff
@@ -1,20 +1,239 @@
...
+}
+
+.live-notifications-denied__header,
+.live-notifications-explainer__header {
+    position: relative;
+    padding-right: gs-span(1) + $gs-gutter * .5;
+
+    .inline-marque-36 {
+        position: absolute;
+        left: 0 - $live-notifications-icon-width - $gs-gutter / 2;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Why do you start with `0 -` ?

Comment (sndrs) [inline static/src/stylesheets/module/preferences/_notifications.scss]: readability?

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 18 above)
Comment (NathanielBennett) [inline static/src/stylesheets/module/preferences/_notifications.scss]: OK, I'll change it :)

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 25 above)
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: I'd disagree but I guess it's subjective... :)

File: static/src/javascripts/projects/common/views/ui/notifications-explainer.html (hunk 12 above)
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/views/ui/notifications-explainer.html]: I nicked it from somewhere else .

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 24 above)
Comment (NathanielBennett) [inline static/src/stylesheets/module/preferences/_notifications.scss]: @gtrufitt Looks like you do ( it looks wrong when I take it out )

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 25 above)
Comment (NathanielBennett) [inline static/src/stylesheets/module/preferences/_notifications.scss]: I agree that it's subjective but I think - $live-notifications-icon-width - $gs-gutter / 2; is less readable. So tough

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 24 above)
Comment (sndrs) [inline static/src/stylesheets/module/preferences/_notifications.scss]: think that should be `@include fs-headline(2);`, which is `400` by default

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 26]
```diff
@@ -43,37 +48,65 @@ define([
...
-                    className: hasNoSubscriptions ? '' : 'notifications-subscribe-link--has-subscriptions',
-                    text: subscribed ? 'Unfollow' : 'Follow story',
-                    imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'notification-explainer']),
-                    imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'notification-explainer'])
+                src = template(followLink, {
+                    subscribed: subscribed,
+                    icon: svgs(subscribed ? 'notificationsOff' : 'notificationsOn')
                 });
 
+            $follow.html(src);
```
Comment (OliverJAsh) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: This should be in a fastdom.write

File: static/src/javascripts/bootstraps/enhanced/notifications.js (hunk 11 above)
Comment (OliverJAsh) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Here we are catching all rejected promises and then checking if permissions for notifications have been denied.

What if the error is not related to permissions? It would silently fail? Could you instead look at the type of error that is coming through, and use that to decide whether it is a permissions error? If it's not a permissions error you can rethrow the error (as you don't want to handle it here).

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 24 above)
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: @sndrs Good catch!

File: static/src/javascripts/projects/common/views/ui/notifications-follow-link.html [hunk 27]
```diff
@@ -0,0 +1,7 @@
+ <button class="button button--large button--tertiary js-notifications__button notifications__button notifications-follow-input--solo notifications__button--<%= subscribed ? 'subscribed' : 'unsubscribed' %>">
+    <%= icon %>
+    <span class="live-notifications__label live-notifications__label--unsubscribed">Get alerts for this story</span>
+    <span class="live-notifications__label live-notifications__label--subscribed">Alerts on this story</span>
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/views/ui/notifications-follow-link.html]: Instead of using CSS, you could pass in some state to the template to decide which label to show:

``` html
<% if (isSubscribed) { %>
    <span class="live-notifications__label live-notifications__label--subscribed">Alerts on this story</span>
<% } else { %>
    <span class="live-notifications__label live-notifications__label--unsubscribed">Get alerts for this story</span>
<% } %>
```

File: static/src/javascripts/projects/common/views/ui/notifications-permission-denied-message.html [hunk 28]
```diff
@@ -0,0 +1,10 @@
+<div class="js-live-notifications-denied live-notifications-denied">
+    <div class="live-notifications-denied__content">
+        <div class="live-notifications-denied__header">Looks like you've blocked Notifications! If you've changed your mind, click the lock to control your notifications</div>
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/views/ui/notifications-permission-denied-message.html]: Why is notifications capitalised?

File: static/src/stylesheets/module/preferences/_notifications.scss (hunk 16 above)
Comment (OliverJAsh) [inline static/src/stylesheets/module/preferences/_notifications.scss]: We should be using pixel values in our CSS. They are automatically converted to em at compile time

File: static/src/javascripts/projects/common/views/ui/notifications-permission-denied-message.html (hunk 28 above)
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/views/ui/notifications-permission-denied-message.html]: @OliverJAsh Typo. Thanks

**Comment by janua** [comment]
@NathanielBennett  Is the location of the message for when notifications are blocked always going to be accurately placed?

**Comment by NathanielBennett** [comment]
@janua It was. But now it's pixellated, |'ll check. Thanks!

**Comment by NathanielBennett** [comment]
@gtrufitt @sndrs @OliverJAsh Thanks for the clientside love. I think I've fixed everything now. Can I have a +1?
Thanks!

**Comment by gtrufitt** [comment]
@NathanielBennett Looks like there's a problem with your SVG which is causing a failed build

File: static/src/javascripts/bootstraps/enhanced/liveblog.js [hunk 29]
```diff
@@ -68,9 +68,12 @@ define([
         },
 
         initNotifications: function() {
+/*
             if ((ab.isInVariant('LiveBlogChromeNotificationsInternal', 'control') || ab.isInVariant('LiveBlogChromeNotificationsProd', 'show-notifications') )
                 && (window.location.protocol === 'https:' ||  window.location.hash === '#force-sw')
                 && detect.getUserAgent.browser === 'Chrome' && config.page.isLive) {
+*/
+            if(detect.getUserAgent.browser === 'Chrome'){
```
Comment (gtrufitt) [inline static/src/javascripts/bootstraps/enhanced/liveblog.js]: Would probably be a good idea to add a switch so we can turn this off quickly if need be?

File: static/src/javascripts/projects/common/views/ui/notifications-subscribe-link.html [hunk 30]
```diff
@@ -1,6 +1,6 @@
 <div class="js-notification-link notifications-link">
     <div class="notifications-subscribe-link--follow  <%= className %>">
-        <h3 class="notifications-subscribe-link__headline">Follow the story and stay updated</h3>
+        <div class="notifications-subscribe-link__headline">Follow the story and stay updated</div>
```
Comment (gtrufitt) [inline static/src/javascripts/projects/common/views/ui/notifications-subscribe-link.html]: I would say that a heading was the right mark-up semantically here, why did you change it to a div?

File: static/src/stylesheets/base/_zindex.scss [hunk 31]
```diff
@@ -8,3 +8,4 @@ $zindex-sticky: 1020;  // Sticky header
 $zindex-modal: 1030;   // Search modal
 $zindex-popover: 1040; // Breaking news
 $zindex-overlay: 1050; // Lightbox
+$zindex-notifications-permissions-warning: 15000;
```
Comment (gtrufitt) [inline static/src/stylesheets/base/_zindex.scss]: Is there a z-index somewhere else that caused you to use `15000`? If not, and you're just trying to up a level (I think you just want this to appear above everything), then `1060` should work here.

File: static/src/javascripts/projects/common/views/ui/notifications-follow-link.html (hunk 27 above)
Comment (gtrufitt) [inline static/src/javascripts/projects/common/views/ui/notifications-follow-link.html]: Think you missed this?

File: static/src/stylesheets/module/preferences/_notifications.scss [hunk 32]
```diff
@@ -1,20 +1,248 @@
...
+
+.live-notifications-explainer__header-blurb {
+    @include fs-textSans(3);
+    margin-top: $gs-baseline /2;
+}
+
+.live-notifications-explainer__headline {
+    @include fs-headline(2);
+    color: #ffffff;
+    font-weight: 500;
```
Comment (gtrufitt) [inline static/src/stylesheets/module/preferences/_notifications.scss]: Shouldn't need to override the font-weight now you've fixed `fs-headline`?

File: static/src/javascripts/projects/common/views/ui/notifications-subscribe-link.html (hunk 30 above)
Comment (NathanielBennett) [inline static/src/javascripts/projects/common/views/ui/notifications-subscribe-link.html]: This isn't being used any more anyway ..

Comment (gtrufitt) [inline static/src/javascripts/projects/common/views/ui/notifications-subscribe-link.html]: Can you remove it?

**Comment by gtrufitt** [comment]
👍 
8a4f5383c45d3960a60c14445fe627e1

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 33]
```diff
@@ -43,37 +48,69 @@ define([
...
         },
 
         subscribeHandler: function () {
-            modules.subscribe().then(modules.follow);
+            modules.subscribe().then(modules.follow)
+                .catch( function () {
+                    if (Notification.permission === 'denied') {
+                        modules.notificationsDeniedMessage();
+                    }
+                });
```
Comment (OliverJAsh) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: We should be throwing errors we don't care about, otherwise things will silently fail. It's not safe to presume any error that arrives here will be permissions related.

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 34]
```diff
@@ -43,37 +48,69 @@ define([
...
+                imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'live-notifications-explainer-svg']),
+                imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'live-notifications-explainer-svg'])
+            });
+
             fastdom.write(function () {
-                $('.js-notifications').prepend(src);
-                bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+                var $notifications = $('.js-live-notifications');
+                $notifications.append(src);
+               // bean.one($notifications[0], 'click', '.js-live-notifications__item__close', function() {
```
Comment (OliverJAsh) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Remove this comment?
